lib/resty/iprepd.lua

```diff
@@ -191,17 +199,39 @@ end
 
 function _M.audit_log(self, ip)
   if self.audit_blocked_requests == 1 then
-    ngx.req.read_body()
+    local should_audit_request = 0
     local headers = ngx.req.get_headers()
-    local request_headers_all = ""
-    for k,v in pairs(headers) do
-      request_headers_all = request_headers_all .. string.format('"%s": "%s", ', k, v)
+    for i, uri in ipairs(self.audit_uri_list) do
+      if string.match(ngx.var.uri, uri) then
+        should_audit_request = 1
+        break
+      end
     end
-    local data = ngx.req.get_body_data()
-    if not data then
-      data = ngx.req.get_body_file()
+    if should_audit_request == 1 then
+      ngx.req.read_body()
+      local request_headers_all = ""
+      if self.audit_include_headers == 1 then
+        local headers = ngx.req.get_headers()
+        for k,v in pairs(headers) do
+          if k == "authorization" or k == "proxy-authorization" then
+            v = "REMOVED"
+          end
+          request_headers_all = request_headers_all .. string.format('"%s": "%s", ', k, v)
+        end
+      end
+      request_headers_all = ngx.encode_base64(request_headers_all)
+      local data = ngx.req.get_body_data()
+      if data then
+        data = ngx.encode_base64(data)
```

Comment:
Why would we only encode the body if we got it from `ngx.req.get_body_data()`? Does `ngx.req.get_body_file()` return something different?

Reply:
Yes. If the body is less than `client_body_buffer_size` than `ngx.req.get_body_data()` returns nil and `ngx.req.get_body_file` will return the path to the file where the body was cached.